### PR TITLE
[ RMSProp ] add rms prop optimizer

### PR DESCRIPTION
**Pull request by jijoongmoon**
## In this PR
This PR implements rms prop optimizer.
unittest for rms prop is required.
. implementation of RMSProp Class
. Separate the optimizer properties into optimizer common properties. and some fixes.
. add enum for rms props

**Self evaluation:**
1. Build test:	 [X]Passed [ ]Failed [ ]Skipped
2. Run test:	 [X]Passed [ ]Failed [ ]Skipped

### CODE REVIEW

File: nntrainer/optimizers/rmsprop.cpp [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+  Tensor &wv = context.getOptimizerVariable(RMSPropParams::wv);
+
+  wv.multiply_i(rho);
+  wv.add_i(x_grad.multiply(x_grad), 1.0f - rho);
+
+  if (torch_ref) {
+    Tensor denom = wv.apply<float>(sqrtFloat<float>);
+    denom.add_i(epsilon);
+
+    wv.divide(denom, x_grad);
```
Comment (EunjuYang) [inline nntrainer/optimizers/rmsprop.cpp:86]: Original code seems  to compute
`W = W - alpha * wv / (sqrt(wv) + epsilon)`, which should be
`W = W - alpha * x_grad / (sqrt(wv) + epsilon)` to be properly applied.
```suggestion
    x_grad.divide_i(denom);
```

**Review by DonghakPark — APPROVED** [review]
I think the CI-related error is a rebase issue. In my local environment, when I built with 2584 as the base, it passed build & test without any problems. Other than that, LGTM!!

File: nntrainer/optimizers/rmsprop.cpp [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+    context.applyGradient(context.getLearningRate(), x_grad);
+
+  } else {
+    std::function<double(double)> sqrtEps = [epsilon](double f) {
+      return 1 / (sqrtDouble(f) + epsilon);
+    };
+
+    x_grad = wv.apply<float>(sqrtEps, x_grad);
+    context.applyGradient(context.getLearningRate(), x_grad);
```
Comment (lhs8928) [inline nntrainer/optimizers/rmsprop.cpp:96]: RumOptimizerContext does not have a function which has a prototype of applyGradient(double, tensor) as the CI issued.

File: nntrainer/optimizers/rmsprop.cpp (hunk 1 above)
Comment (lhs8928) [reply nntrainer/optimizers/rmsprop.cpp:86]: It seems eunju is right.
Further more unlike the adam optimizer torch seems to work in the same way as other frameworks do.
Please refer https://pytorch.org/docs/stable/generated/torch.optim.RMSprop.html

**Review by lhs8928 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**